src/lib/db-listeners/db-publications-listener.ts

```diff
@@ -54,6 +54,22 @@ export class DBPublicationsListener {
       this.db.createDocumentFromPublication(publication)
         .then(doc => {
           documents.add(doc);
+          onPatch(doc.stars, patch => {
+            const path = patch.path.split("/");
+            if (patch.op === "add") {
+              const star = patch.value;
+              const { starred, key } = star;
+              if (key === "") { // key will be assigned from Firebase node id
```

Comment:
Is this `key` check necessary? I can't recall if keeping `key` in the model was necessary at all, even though I know I left it in comments...

Reply:
It is.  Otherwise we get echos when we first construct the document (we enter the onPatch with an add operation and attempt to add the star again).  The empty key allows us to tell that we have a newly created star that hasn't been assigned a Firebase id.  This lets us distinguish between adding a new star and constructing the document on load.

Reply:
Got it - I see why this wasn't necessary in my code path now, too, because of the extra map layer. Using the key to keep track of new stars seems a little unintuitive, but it's also a pretty simple solution so I'm good with it 👍

Reply:
Yeah, I see your point.  On the other hand, I wasn't a fan of generating a key and then overwriting it with a value from Firebase.  That felt wrong to me (and I believe thisis still happening in the comments case).  Seemed natural to let the key originate from Firebase.  Anyway, on the same note,, we use the `if (text)` check in the comments case which also is probably less than ideal.  We should probably rethink both of these cases if we end up having to modify the `onPatch` again (if, for example, we add graph annotations).